From e72e6f1696de60065a777b400cc627f27c132c73 Mon Sep 17 00:00:00 2001
From: Roman Birg <roman@cyngn.com>
Date: Thu, 22 Feb 2018 15:27:46 +0100
Subject: [PATCH 21/26] Settings: Add LineageOS legal info

Open up the LineageOS legal info in the browser.

Change-Id: I263ccc0509e275d17512528deb606341d58e7a0d
Ticket-Id: CYNGNOS-1895
Signed-off-by: Roman Birg <roman@cyngn.com>
---
 res/values/cm_strings.xml                   |  3 +++
 res/xml/about_legal.xml                     |  5 +++++
 src/com/android/settings/LegalSettings.java | 24 +++++++++++++++++++++
 3 files changed, 32 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 8e905993c0..7fa198c702 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -54,6 +54,9 @@
     <string name="kill_app_longpress_back">Kill app back button</string>
     <string name="kill_app_longpress_back_summary">Kill the foreground app by long-pressing the back button</string>
 
+    <!-- LineageOS legal -->
+    <string name="lineagelicense_title">LineageOS legal</string>
+
     <!-- Lock screen cover art -->
     <string name="lockscreen_media_art_title">Display media cover art</string>
 
diff --git a/res/xml/about_legal.xml b/res/xml/about_legal.xml
index 55faad316d..ee3bc635bb 100644
--- a/res/xml/about_legal.xml
+++ b/res/xml/about_legal.xml
@@ -37,6 +37,11 @@
         <intent android:action="android.settings.LICENSE"/>
     </Preference>
 
+    <!-- Lineage License information -->
+    <PreferenceScreen
+        android:key="lineagelicense"
+        android:title="@string/lineagelicense_title" />
+
     <!-- Terms and conditions -->
     <Preference
         android:key="terms"
diff --git a/src/com/android/settings/LegalSettings.java b/src/com/android/settings/LegalSettings.java
index e9b2694b82..398dc95af8 100644
--- a/src/com/android/settings/LegalSettings.java
+++ b/src/com/android/settings/LegalSettings.java
@@ -22,10 +22,15 @@ import android.content.Intent;
 import android.content.pm.ApplicationInfo;
 import android.content.pm.PackageManager;
 import android.content.pm.ResolveInfo;
+import android.net.Uri;
 import android.os.Bundle;
+import android.os.SystemProperties;
 import android.provider.SearchIndexableResource;
 import android.support.annotation.VisibleForTesting;
+import android.support.v7.preference.Preference;
 import android.support.v7.preference.PreferenceGroup;
+import android.support.v7.preference.PreferenceScreen;
+import android.util.Log;
 
 import com.android.internal.logging.nano.MetricsProto.MetricsEvent;
 import com.android.settings.R;
@@ -37,11 +42,14 @@ import java.util.List;
 
 public class LegalSettings extends SettingsPreferenceFragment implements Indexable {
 
+    private static final String LOG_TAG = "LegalSettings";
     private static final String KEY_TERMS = "terms";
     private static final String KEY_LICENSE = "license";
     private static final String KEY_COPYRIGHT = "copyright";
     private static final String KEY_WEBVIEW_LICENSE = "webview_license";
     @VisibleForTesting static final String KEY_WALLPAPER_ATTRIBUTIONS = "wallpaper_attributions";
+    private static final String PROPERTY_LINEAGELICENSE_URL = "ro.lineagelegal.url";
+    private static final String KEY_LINEAGE_LICENSE = "lineagelicense";
 
     public void onCreate(Bundle icicle) {
         super.onCreate(icicle);
@@ -62,6 +70,22 @@ public class LegalSettings extends SettingsPreferenceFragment implements Indexab
         checkWallpaperAttributionAvailability(act);
     }
 
+    @Override
+    public boolean onPreferenceTreeClick(Preference preference) {
+        if (preference.getKey().equals(KEY_LINEAGE_LICENSE)) {
+            String userLineageLicenseUrl = SystemProperties.get(PROPERTY_LINEAGELICENSE_URL);
+            final Intent intent = new Intent(Intent.ACTION_VIEW);
+            intent.addCategory(Intent.CATEGORY_DEFAULT);
+            intent.setData(Uri.parse(userLineageLicenseUrl));
+            try {
+                startActivity(intent);
+            } catch (Exception e) {
+                Log.e(LOG_TAG, "Unable to start activity " + intent.toString());
+            }
+        }
+        return super.onPreferenceTreeClick(preference);
+    }
+
     @Override
     public int getMetricsCategory() {
         return MetricsEvent.ABOUT_LEGAL_SETTINGS;
-- 
2.17.1

